persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java

```diff
@@ -159,22 +162,23 @@ public void initialize() {
           "Unable to intercept proxy protocol system query without a valid public address");
     }
     String tableName = selectStatement.table();
+    Set<InetAddress> currentPeers = clientState.isInternal() ? internalPeers : externalPeers;
```

Comment:
Instead of having an "internal" flag could we check the peers for membership? This is more generalized, and we could potential support more than two sets (internal, external). I'm thinking about something like supporting private link + VPC peering + public, etc.

Reply:
That would work too, but is there any risk that the peer sets could be out of sync (with the refresh interval being 10 seconds). We'd have to error out if we don't find the address anywhere.

The current implementation is immune to that because:
* system.peers = knownPeers - publicAddress
* system.local = publicAddress (does not check that it's present in knownPeers)